### PR TITLE
[New] Migrating from k8s to Terraform

### PR DESCRIPTION
**Pull request by rsyracuse**
Survival guide for k8s deprecation

### CODE REVIEW

**Comment by nmelehan** [comment]
Deploy preview for *nostalgic-ptolemy-b01ab8* ready!

Built with commit a0e2deee0ac7e54be1a8221bfe630d4b87c78fae

https://deploy-preview-3055--nostalgic-ptolemy-b01ab8.netlify.com

File: docs/kubernetes/how-to-migrate-from-k8s-alpha-to-terraform/index.md [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+mycluster-node-1 	Ready	 &lt;none&gt;   18m 	  v1.13.6
+mycluster-node-2 	Ready	 &lt;none&gt;   18m 	  v1.13.6
+mycluster-node-3 	Ready	 &lt;none&gt;   18m 	  v1.13.6
+mycluster-node-4 	Ready	 &lt;none&gt;   4m26s   v1.13.6
+mycluster-node-5 	Ready	 &lt;none&gt;   4m52s   v1.13.6
+{{< /output >}}
+
+### Upgrade a Cluster
+
+You may have noticed that the Terraform module, `cluster.tf`, refers to a specific branch and git commit hash referencing a remote Terraform module on GitHub. The following section will outline how to upgrade your cluster to the latest version of this module.
```
[the remark this answers is not in the document]
Comment (hzoppetti) [reply docs/kubernetes/how-to-migrate-from-k8s-alpha-to-terraform/index.md]: updated

File: docs/kubernetes/how-to-migrate-from-k8s-alpha-to-terraform/index.md [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+- `.terraform` is a hidden directory which contains Terraform configuration files.
+- `cluster.tf` is the Terraform module file. This is the most important file here because it will allow you to scale, upgrade, and delete your cluster.
+
+{{< note >}}
+For more information regarding these files and directories and their contents, see our [Beginner's Guide to Terraform](/applications/configuration-management/beginners-guide-to-terraform/)
+{{< /note >}}
+
+### Scale a Cluster
+
+1.  To scale your cluster, edit `cluster.tf` with the text editor of your choice. The contents will be similar to the following:
```
Comment (andystevensname) [inline docs/kubernetes/how-to-migrate-from-k8s-alpha-to-terraform/index.md]: This line has the same beginning as step 2: "To scale your cluster..." One should change, and I think it's this one.

Comment (hzoppetti) [reply docs/kubernetes/how-to-migrate-from-k8s-alpha-to-terraform/index.md]: updated

File: docs/kubernetes/how-to-migrate-from-k8s-alpha-to-terraform/index.md [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+mycluster-node-1 	Ready	 &lt;none&gt;   18m 	  v1.13.6
+mycluster-node-2 	Ready	 &lt;none&gt;   18m 	  v1.13.6
+mycluster-node-3 	Ready	 &lt;none&gt;   18m 	  v1.13.6
+mycluster-node-4 	Ready	 &lt;none&gt;   4m26s   v1.13.6
+mycluster-node-5 	Ready	 &lt;none&gt;   4m52s   v1.13.6
+{{< /output >}}
+
+### Upgrade a Cluster
+
+You may have noticed that the Terraform module, `cluster.tf`, refers to a specific branch or git commit hash referencing a remote Terraform module on GitHub. The following section will outline how to upgrade your cluster to the latest version of this module.
```
Comment (andystevensname) [inline docs/kubernetes/how-to-migrate-from-k8s-alpha-to-terraform/index.md]: Do we know what this referenced module does? Just wondering if it's important to explain it to a user.

File: docs/kubernetes/how-to-migrate-from-k8s-alpha-to-terraform/index.md [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+5e68ff7beee9c36aa4a4f5599f3973f753b1cd9e
+{{< /output >}}
+
+1.  Edit `cluster.tf` to prepare for the upgrade. Update the following section using the hash you copied to appear as follows:
+
+    {{< file "cluster.tf" >}}
+source = "git::https://github.com/linode/terraform-linode-k8s.git?ref=5e68ff7beee9c36aa4a4f5599f3973f753b1cd9e"
+{{< /file >}}
+
+1.  To apply these changes, re-init the modules by running the following command:
```
Comment (andystevensname) [inline docs/kubernetes/how-to-migrate-from-k8s-alpha-to-terraform/index.md]: re-initialize

File: docs/kubernetes/how-to-migrate-from-k8s-alpha-to-terraform/index.md [hunk 5]
```diff
@@ -0,0 +1,257 @@
...
+
+        terraform init
+
+1.  Once this has completed, run terraform apply:
```
Comment (andystevensname) [inline docs/kubernetes/how-to-migrate-from-k8s-alpha-to-terraform/index.md]: Once this has completed, apply your changes with the following command: (just trying to remove the command from the sentence)

File: docs/kubernetes/how-to-migrate-from-k8s-alpha-to-terraform/index.md [hunk 6]
```diff
@@ -0,0 +1,257 @@
...
+
+        terraform apply
+
+    {{< note >}}
+Depending on the changes that have been configured, you may or may not see the upgrade perform actions. For example, in this case, because the only change was to the kubernetes version, no actions were taken.
```
Comment (andystevensname) [inline docs/kubernetes/how-to-migrate-from-k8s-alpha-to-terraform/index.md]: capital Kubernetes

File: docs/kubernetes/how-to-migrate-from-k8s-alpha-to-terraform/index.md [hunk 7]
```diff
@@ -0,0 +1,257 @@
...
+To destroy a cluster, navigate to the directory containing your cluster's files, and enter the `terraform destroy` command:
+
+    cd ~/.k8s-alpha-linode/mycluster
+    terraform destroy
+
+Terraform will prompt you to confirm the action, and on confirmation will proceed to destroy all associated resources. If this process is interrupted for any reason, you can run the command again at any time to complete the process.
+
+## Create a Cluster
+
+1.  Create a new directory to house its configuration files in the `~/.k8s-alpha-linode` directory. In this example the cluster name is `mynewcluster`:
```
Comment (andystevensname) [inline docs/kubernetes/how-to-migrate-from-k8s-alpha-to-terraform/index.md]: to house the new cluster's configuration files

File: docs/kubernetes/how-to-migrate-from-k8s-alpha-to-terraform/index.md (hunk 3 above)
Comment (hzoppetti) [reply docs/kubernetes/how-to-migrate-from-k8s-alpha-to-terraform/index.md]: ok I changed this and updated the first paragraph a little too to be more clear about what this actually is.

File: docs/kubernetes/how-to-migrate-from-k8s-alpha-to-terraform/index.md (hunk 4 above)
Comment (hzoppetti) [reply docs/kubernetes/how-to-migrate-from-k8s-alpha-to-terraform/index.md]: done

File: docs/kubernetes/how-to-migrate-from-k8s-alpha-to-terraform/index.md (hunk 5 above)
Comment (hzoppetti) [reply docs/kubernetes/how-to-migrate-from-k8s-alpha-to-terraform/index.md]: updated

File: docs/kubernetes/how-to-migrate-from-k8s-alpha-to-terraform/index.md (hunk 6 above)
Comment (hzoppetti) [reply docs/kubernetes/how-to-migrate-from-k8s-alpha-to-terraform/index.md]: done

File: docs/kubernetes/how-to-migrate-from-k8s-alpha-to-terraform/index.md (hunk 7 above)
Comment (hzoppetti) [reply docs/kubernetes/how-to-migrate-from-k8s-alpha-to-terraform/index.md]: updated

**Pull request merged**